### ISSUE TITLE
wdio is not defined

### ISSUE THREAD

**Issue by pawelos88** [xolvio/meteor-webdriver#2]
Hi,

After adding this package, I get "wdio is not defined", when trying to use the web driver in mocha (velocity) server tests.

**Comment by annachu** [comment]
Hi Sam,

I'm having trouble running some code extracted from your book (in the "Moving between test boundaries" chapter), specifically this piece:

    describe("Email validation", function () {
        
        beforeEach(function() {
            client = wdio.remote({ desiredCapabilities: {browserName: 'phantomjs'} });
            client.init();
            client.url('https://localhost:3000')
        });
        
        it("should respond with Email OK for valid emails", function (done) {
            client.call(done);
        });
        
    });

This returns "wdio is not defined", as expected. But when I tack on `Package['xolvio:webdriver'].wdio` to get `client = Package['xolvio:webdriver'].wdio.remote({ desiredCapabilities: {browserName: 'phantomjs'} });`, I get the error

    Cannot read property 'remote' of undefined

I also get this error for `getGhostDriver()` when I follow the readme.

Am I calling `Package['xolvio:webdriver'].wdio` in the wrong place?

Thanks!

**Comment by samhatoum** [comment]
In the latest version (0.1.7) you should be able to do this to get the phantom browser:

```javascript
      Package['xolvio:webdriver'].wdio.getGhostDriver(function (browser) {
        world.browser = browser;
      });
```

if you want the remove, you'll need to do this:

```javascript
Package['xolvio:webdriver'].wdio.instance.remote(...)
```

**Comment by annachu** [comment]
Hm..I made the changes as suggested, but in both cases, I still get the same error.
For example, after updating to `Package['xolvio:webdriver'].wdio.instance.remote(...)`, I get

    Cannot read property 'instance' of undefined

I am using version 0.1.7.

**Comment by samhatoum** [comment]
that's really odd. could you please create a simple reporo repo and I can look

**Comment by annachu** [comment]
Sure! Please see https://github.com/annachu/webdriver-test

And in case this helps, here are the steps I took
1. Created a new meteor project
2. Added sanjo:jasmine
3. Added velocity:html-reporter
4. Added xolvio:webdriver
5. Clicked on "Add jasmine-client-integration sample tests" in the Velocity overlay
6. Removed the sample files from `tests/jasmine/client/integration/sample`
7. Commented out the corresponding `api.add_files()` lines in `packages/tests-proxy/package.js`
8. Added the file `tests/jasmine/client/integration/sample/webdriver-test.js`
9. Restarted meteor
10. Went to localhost:3000
11. Received error `Cannot read property 'instance' of undefined`

**Comment by samhatoum** [comment]
oh, that's easy! change the test to a server test instead of client :)

webdriver is a server only package. I should update the readme.

**Comment by samhatoum** [comment]
it uses phantom, which runs on the server portion

**Comment by samhatoum** [comment]
so just rename:

`webdriver-test/tests/jasmine/client/integration/sample/`

to 

`webdriver-test/tests/jasmine/server/integration/sample/`

and I think you're good to go

**Comment by annachu** [comment]
Hey Sam - so I tried renaming and got a cordova? error...
I built a new project from scratch and went straight to 'server integration' tests to make sure that it wasn't just me contaminating things in the move :(

    /Users/anna/.meteor/packages/meteor-tool/.1.0.38.16ro34c++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:173
                            throw(ex);
                                  ^
    ReferenceError: describe is not defined
      at Package (/Users/anna/Documents/github/test/.meteor/local/build/programs/server/packages/velocity_test-proxy.js:184:1)
      at /Users/anna/Documents/github/test/.meteor/local/build/programs/server/packages/velocity_test-proxy.js:200:4
      at /Users/anna/Documents/github/test/.meteor/local/build/programs/server/packages/velocity_test-proxy.js:207:3
      at /Users/anna/Documents/github/test/.meteor/local/build/programs/server/boot.js:175:10
      at Array.forEach (native)
      at Function._.each._.forEach (/Users/anna/.meteor/packages/meteor-tool/.1.0.38.16ro34c++os.osx.x86_64+web.browser+web.cordova/meteor-tool-os.osx.x86_64/dev_bundle/server-lib/node_modules/underscore/underscore.js:79:11)
      at /Users/anna/Documents/github/test/.meteor/local/build/programs/server/boot.js:86:5

    Exited with code: 8
    Your application is crashing. Waiting for file change.

**Comment by annachu** [comment]
And as for phantom on the server side...oops and duh :P  Thanks for catching that!

**Comment by annachu** [comment]
I don't know if this helps... but I just looked at cucumber and saw that it uses webdriver. So I went to run the leaderboard example in the velocity-examples repo.

When I ran the example as-is, everything was fine. But I saw the message saying that I should upgrade to Meteor 1.0.2.  When I did, I received the following error

    Exception in callback of async function: Error: ENOENT, no such file or directory '/Users/anna/Documents/github/velocity-examples/leaderboard-cucumber/.meteor/local/build/programs/server/assets/packages/xolvio_webdriver/libs/spawner.js'

Could this have something to do with the latest version of Meteor?

**Comment by samhatoum** [comment]
not sure yet, I'll try to find some time this week/weekend to look at this

**Comment by annachu** [comment]
Cool - thank you! It's amazing how quickly you get back to me on these questions. I know you're super busy!

**Comment by samhatoum** [comment]
Fixed in 0.1.8 - let me know and please open a new issue if there is anything. Thanks!

**Comment by annachu** [comment]
Thanks very much! I will give this a try later.
